xrpl/models/ledger_objects/ledger_object.py

```diff
@@ -0,0 +1,189 @@
+
+        Raises:
+            XRPLModelException: If there is no Ledger Object type is
+                provided or the type mismatches the constructor type.
+        """
+        if cls.__name__ == "LedgerObject":
+            if "ledger_entry_type" not in value:
+                raise XRPLModelException(
+                    "Ledger Object does not include ledger_entry_type."
+                )
+            correct_type = cls.get_ledger_object_type(value["ledger_entry_type"])
+            return correct_type.from_dict(value)  # type: ignore
+        elif cls.__name__ in ["FinalFields", "PreviousFields", "NewFields"]:
+            if "ledger_entry_type" not in value:
+                raise XRPLModelException(
+                    "Ledger Object does not include ledger_entry_type."
+                )
+            correct_type = cls.get_md_ledger_object_type(value["ledger_entry_type"])
+            del value["ledger_entry_type"]
+            return correct_type.from_dict(value)  # type: ignore
+        else:
+            if "ledger_entry_type" in value:
+                ledger_entry_type = value["ledger_entry_type"]
+                if (
+                    cls.get_ledger_object_type(ledger_entry_type).__name__
+                    != cls.__name__
+                    and cls.get_md_ledger_object_type(ledger_entry_type).__name__
+                    != cls.__name__
+                ):
+                    raise XRPLModelException(
+                        f"Using wrong constructor: using {cls.__name__} constructor "
+                        f"with ledger object type {value['ledger_entry_type']}."
+                    )
+                value = {**value}
+                del value["ledger_entry_type"]
+            return super(LedgerObject, cls).from_dict(value)
+
+    @classmethod
+    def get_ledger_object_type(
+        cls: Type[LedgerObject], ledger_object_type: str
+    ) -> Type[LedgerObject]:
+        """Get the correct model
+
+        Args:
+            ledger_object_type: The ledger object type willing to get.
+
+        Returns:
+            Type[AffectedNode]: The correct ledger object type.
+
+        Raises:
+            XRPLModelException: If the Ledger Object type does not exist.
+        """
+        import xrpl.models.ledger_objects as ledger_object_models
+
+        ledger_object_types: Dict[str, Type[LedgerObject]] = {
+            lgr_obj.value: getattr(ledger_object_models, lgr_obj)
+            for lgr_obj in ledger_object_models.ledger_entry_type.LedgerEntryType
+        }
+        if ledger_object_type in ledger_object_types:
+            return ledger_object_types[ledger_object_type]
+
+        raise XRPLModelException(
+            f"{ledger_object_type} is not a valid Ledger Object type."
+        )
+
+    @classmethod
+    def get_md_ledger_object_type(
+        cls: Type[LedgerObject], ledger_object_type: str
+    ) -> Type[LedgerObject]:
+        """
+        Get the correct model
```

Comment:
Can we provide a more verbose description for this function, in order to differentiate from the previous function? Ex: Returns ledger object types suffixed with "Fields" and prefixed with "MD" (or something like that?)